src/Core/Repositories/EntityFramework/UserRepository.cs

```diff
@@ -23,7 +23,8 @@ public async Task<TableModel.User> GetByEmailAsync(string email)
             using (var scope = ServiceScopeFactory.CreateScope())
             {
                 var dbContext = GetDatabaseContext(scope);
-                return await GetDbSet(dbContext).FirstOrDefaultAsync(e => e.Email == email);
+                var entity = await GetDbSet(dbContext).FirstOrDefaultAsync(e => e.Email == email);
```

Comment:
Are you considering case-sensitivity of various DB collations of mySQL and/or PostgreSQL to ensure value matches on strings like this are either case-insensitive OR data is normalized on the input to the DB via the provider? Our SQL Server configuration, for example, uses a `_CI` collation so this check would pass regardless of case, but other DBs may not be the same.

Reply:
This was a fun one to address.

No, I was not considering collation, and Postgres has some hoops to jump through in that regard.
1. There is no option for a database-level collation rule to be applied.
2. `LIKE` clauses don't work for case-insensitive collations.

To handle these I:
1. Created a case-insensitive collation [here](https://github.com/bitwarden/server/pull/1265/files#diff-5b19f4a346a42e5bf955ea7b5ebb0fea518559757ac96f08636ee88d298700edR80-R86) and applied it to any string columns that are being used to look up entities.
2. For methods that use `LIKE` searches on fields that would typically use case-insensitive collations I added some provider-specific logic (😔 ) for postgres to use the default collation & their `ILIKE` function that is case-insensitive.